bench/bench.py

```diff
@@ -0,0 +1,189 @@
+import statistics
+import time
+
+
+class Bench:
+    name = None
+    driver = None
+    queries = []
+
+    def __call__(self):
+        try:
+            self.setup()
+        except ImportError:
+            print(f'Driver {self.driver} not installed, skipping…')
+            return
+
+        for query in self.queries:
+            self.time(query)
+
+        self.close()
+
+    def time(self, query, iterations=10):
+        times = [None] * iterations
+
+        print(query)
+        for i in range(iterations):
+            start = time.perf_counter()
+            self.run(query)
+            times[i] = time.perf_counter() - start
+
+        median = statistics.median(times)
+        mean = statistics.mean(times)
+        max_ = max(times)
+        min_ = min(times)
+        print(
+            f"Mean: {mean:.4f} - "
+            f"Median: {median:.4f} - "
+            f"Max: {max_:.4f} - "
+            f"Min: {min_:.4f}"
```

Comment:
Is there a reason you limit to 4 digits ? Because you end up with a single significant digit, and it looks not very useful for comparing.

If you think the precision of perf_counter() cannot be trusted under this value, maybe it would be worth running each query 10, 100 or 1000 times, and then repeating this n times for statistics as you already do (except if you think there might be caching on the PG side)

Reply:
This was just to be easier to read, I want to look at trends not precise numbers. I guess I could use `perf_rec_ns` & use ns as a base unit instead.